awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -556,6 +563,23 @@ def create_if_needed(self, existing_item, new_item, endpoint, on_create=None, it
         else:
             self.exit_json(**self.json_output)
 
+    # We need to be able to recursevly step through fields in the case of inputs for credentials.
+    # They are dicts and we can't just compare them at the top level because the dict returned from tower_cli will have fields like $encrypted$
+    def compare_fields(self, new_item, existing_item):
+        needs_update = False
+        for field in new_item:
+            existing_field = existing_item.get(field, None)
+            new_field = new_item.get(field, None)
+            if type(existing_field) == dict and type(new_field) == dict:
+                needs_update = needs_update or self.compare_fields(new_field, existing_field)
```

Comment:
You're calling `self.compare_fields`, but you don't really need `self`.

for a pure "utility" method like this, it would be better to decorate it with `@staticmethod` to make clear that it doesn't need (and shouldn't have) broad access to the instance data like `self.params`. That would make the recursive call `TowerModule.compare_fields`

Reply:
Updated.